models/template.go

```diff
@@ -9,16 +9,24 @@ import (
 type QuestionType string
 
 const (
-	TextQuestion   QuestionType = "text"
-	NumberQuestion QuestionType = "number"
-	SelectQuestion QuestionType = "select"
+	TextQuestion     QuestionType = "text"
+	NumberQuestion   QuestionType = "number"
+	SelectQuestion   QuestionType = "select"
+	RangeQuestion    QuestionType = "range"
+	TextAreaQuestion QuestionType = "text_area"
+	RadioQuestion    QuestionType = "radio"
+	CheckboxQuestion QuestionType = "checkbox"
 )
 
 type Question struct {
 	ID          primitive.ObjectID `bson:"_id,omitempty"`
 	Title       string             `bson:"title"`
 	Description string             `bson:"description,omitempty"`
 	Type        QuestionType       `bson:"type"`
+	Options     []string           `bson:"options,omitempty"`
+	Min         int                `bson:"min,omitempty"`
+	Max         int                `bson:"max,omitempty"`
+	Order       int                `bson:"order"`
```

Comment:
I added order because i envision us being able to drag and drop questions up and down the template in the future, and we need to be able to determine which order the user saved them in to display later on.

Reply:
Makes sense to me